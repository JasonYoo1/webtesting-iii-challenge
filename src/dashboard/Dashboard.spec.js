// Test away
import React from 'react';
import { render, fireEvent } from '@testing-library/react'


import Dashboard, { asyncFunc } from './Dashboard'

//integration test, texting component itself

describe("<Dashboard />", ()=>{
    it('has the state unlock lock open close', ()=>{
        const { getByText } = render(<Dashboard/>);
        //this is done by checking state in Dashboard.js
        expect(getByText('Unlocked'))
        expect(getByText('Open'));
        expect(getByText('Lock Gate'));
        expect(getByText('Close Gate'));
        
    })
    it('checks buttons if it change state', () => {
        const { getByText } = render(<Dashboard />);
  
        const closeButton = getByText("Close Gate")
        fireEvent.click(closeButton);
        expect(getByText('Unlocked'));
        expect(getByText('Closed'));
        expect(getByText('Lock Gate'));
        expect(getByText('Open Gate'));

        const lockButton = getByText("Lock Gate")
        fireEvent.click(lockButton)
        expect(getByText('Locked'));
        expect(getByText('Closed'));
        expect(getByText('Unlock Gate'));
        expect(getByText('Open Gate'));

        const unlockButton = getByText("Unlock Gate")
        fireEvent.click(unlockButton)
        expect(getByText('Unlocked'));
        expect(getByText('Closed'));
        expect(getByText('Lock Gate'));
        expect(getByText('Open Gate'));

        const openButton = getByText("Open Gate")
        fireEvent.click(openButton)
        expect(getByText('Unlocked'));
        expect(getByText('Open'));
        expect(getByText('Lock Gate'));
        expect(getByText('Close Gate'));

    })
});