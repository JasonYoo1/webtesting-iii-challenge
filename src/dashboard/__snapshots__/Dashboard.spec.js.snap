// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      disabled={true}
      id="control-open-close"
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      disabled={false}
      id="control-lock-unlock"
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
